### PR TITLE
refactor: create explicit event and interest services with separate stores.

### PR DESCRIPTION
**Pull request by nathanielc**
Establishes a pattern of services and stores.

A service is a public API to be consumed by other services in the application.
A store is direct access to a sqlite database.

Services have their own stores. Stores never touch tables that belong to other services.

### CODE REVIEW

**Review by dav1do — APPROVED** [review]
:shipit: Bring on the merge conflicts!

**Review by stbrody — COMMENTED** [review]
nice

**Review by stbrody — APPROVED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -6,22 +6,23 @@ members = [
     "car",
     "core",
     "event",
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply Cargo.toml:8]: The `ceramic-event` crate contains the event types with no behavior other than encoding/decoding them. This is useful for other downstream projects (like potential future clients) that want to leverage the event types without bringing all the dependencies of the `event-svc` (sqlite for example).

File: event-svc/src/store/metrics.rs [hunk 2]
```diff
@@ -123,9 +123,9 @@ impl<S: Send + Sync> StoreMetricsMiddleware<S> {
 }
 
 #[async_trait]
-impl<S> ceramic_api::InterestStore for StoreMetricsMiddleware<S>
+impl<S> ceramic_api::InterestService for StoreMetricsMiddleware<S>
```
Comment (smrz2001) [inline event-svc/src/store/metrics.rs:126]: This is still needed in the Event Service?

File: interest-svc/src/store/metrics.rs [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+            &self.metrics,
+            "api_interest_range",
+            self.store.range(start, end, offset, limit),
+        )
+        .await
+    }
+}
+
+#[async_trait]
+impl<S> ceramic_api::EventService for StoreMetricsMiddleware<S>
```
Comment (smrz2001) [inline interest-svc/src/store/metrics.rs:157]: This is still needed in the Interest Service?

File: event-svc/src/store/metrics.rs (hunk 2 above)
Comment (nathanielc) [reply event-svc/src/store/metrics.rs:126]: nope good catch will remove, and the mirror as well.

**Pull request merged**